packages/vitest/src/node/watcher.ts

```diff
@@ -171,3 +207,11 @@ export class VitestWatcher {
     return !!files.length
   }
 }
+
+export interface WatcherTriggerPattern {
+  pattern: RegExp
+  testsToRun: (
+    file: string,
+    match: RegExpMatchArray
+  ) => string[] | string | null | undefined | void
```

Comment:
Is it possible to make this `async`? I would already have use case for this, where I would be watching for `fixtures/**/source.ts` changes, and then trigger `async` build to transform that into `fixtures/**/dist/transpiled.js` that is loaded by tests.

Reply:
I don't know if it's a good idea to trigger side effects. Maybe it's better to do in `onTestsRerun` in the global setup?

Reply:
Yeah maybe this isn't correct place for side effects.